src/Microsoft.AspNetCore.ResponseCaching/ResponseCacheMiddleware.cs

```diff
@@ -238,16 +238,18 @@ internal void FinalizeCacheHeaders(ResponseCacheContext context)
                 }
 
                 // Ensure date header is set
-                if (context.TypedResponseHeaders.Date == null)
+                if (context.ResponseDate == null)
                 {
-                    context.TypedResponseHeaders.Date = context.ResponseTime;
+                    context.ResponseDate = context.ResponseTime;
+                    // Setting the date on the raw response headers.
+                    context.TypedResponseHeaders.Date = context.ResponseDate;
```

Comment:
Half the point was to remove this line since it is unnecessary and a lot of extra work